new/property-defaults.md

```diff
@@ -0,0 +1,120 @@
+---
+RFC: unassigned
+Author: John Keiser <jkeiser@chef.io>
+Status: Draft
+Type: Standards Track
+---
+
+# Property Default Value Improvements
+
+This RFC addresses some longstanding issues with `default` values on properties (attributes), adds `reset_property` to Resource, and addresses how we will move core Chef resources over to use defaults and properties.
+
+## Motivation
+
+    As a Chef developer,
+    I want the core resources to use `property`,
+    So that people are cribbing off good examples.
+
+    As a Chef user,
+    I want the core resources not to actually change behavior,
+    So that I can rely on things I've built on top of them.
+
+## Specification
+
+### Non-Sticky Defaults
+
+Defaults are presently *sticky*, in that the first time a property is accessed, it is stored in the instance variable:
+
+```ruby
+class X < Chef::Resource
+  attribute :foo, default: lazy { Random.rand }
+end
+
+x 'blah' do
+  puts @foo #=> nil
+  puts foo  #=> 1
+  puts foo  #=> 1
+  puts @foo #=> 1
+end
+```
+
+Since this has some utility, we add a new keyword for computed defaults.
+
+#### Computed defaults
+
+```ruby
```

Comment:
Kind of meh on this idea. Adding a new keyword to the API for properties isn't terrible, but it is already a big API so just noting that we should proceed cautiously. The added functionality isn't really something I've ever seen the need for, generally I would use a method when I want something not cached and based on other properties. I can see some argument in very complex cases though so I wouldn't fight it specifically. Docs should make it clear this is a rare case and probably not what a new user wants (they want `lazy`).